internal/pkg/cli/deploy/workload.go

```diff
@@ -311,40 +353,52 @@ func (d *workloadDeployer) uploadContainerImage(imgBuilderPusher imageBuilderPus
 		return nil, nil
 	}
 	// If it is built from local Dockerfile, build and push to the ECR repo.
-	buildArg, err := buildArgs(d.name, d.imageTag, d.workspacePath, d.mft)
+	buildArg, err := buildArgs(d.name, d.workspacePath, d.runtimeImage, d.mft)
 	if err != nil {
 		return nil, err
 	}
-	digest, err := imgBuilderPusher.BuildAndPush(dockerengine.New(exec.NewCmd()), buildArg)
-	if err != nil {
-		return nil, fmt.Errorf("build and push image: %w", err)
+	images := make(map[string]RuntimeImage, len(buildArg))
+	for k, v := range buildArg {
+		digest, err := imgBuilderPusher.BuildAndPush(dockerengine.New(exec.NewCmd()), v)
+		if err != nil {
+			return nil, fmt.Errorf("build and push image: %w", err)
+		}
+		images[k] = RuntimeImage{
+			Digest:         digest,
+			CustomTag:      d.runtimeImage.CustomTag,
+			GitShortCommit: d.runtimeImage.GitShortCommit,
+			UUID:           d.runtimeImage.UUID,
+		}
 	}
-	return aws.String(digest), nil
+	image := images[d.name]
+	return &image.Digest, nil
 }
 
-func buildArgs(name, imageTag, workspacePath string, unmarshaledManifest interface{}) (*dockerengine.BuildArguments, error) {
+func buildArgs(name, workspacePath string, i RuntimeImage, unmarshaledManifest interface{}) (map[string]*dockerengine.BuildArguments, error) {
```

Comment:
There is a fine balance with naming variables, you want to be as short as possible while not not losing any info (https://a-nickels-worth.dev/papers/varnaming/#headline-12).

This is a matter of taste so feel free to disagree, but `i` in my mind is always associated with "index" and for-loops. Instead, `img` is 2 characters extra but gets rid of that ambiguity when I am reading the body of this function:

```suggestion
func buildArgsPerContainer(name, workspacePath string, img RuntimeImage, unmarshaledManifest interface{}) (map[string]*dockerengine.BuildArguments, error) {
```